File: Python Labs/com/pattern/сhainOfResponsibility/FixedAmountDiscountHandler.py
from com.pattern.сhainOfResponsibility.DiscountHandler import DiscountHandler


class FixedAmountDiscountHandler(DiscountHandler):
    def __init__(self):
        self.next_handler = None

    def set_next_handler(self, next_handler):
        self.next_handler = next_handler

    def apply_discount(self, order):
        if order.total_amount >= 50:
            order.apply_discount(5.0)  # $5 discount
            print("Applied $5 discount.")
        elif self.next_handler is not None:
            self.next_handler.apply_discount(order)

File: Java Labs/Patterns On Java/src/com/pattern/chainOfResponsibility/FixedAmountDiscountHandler.java
package com.pattern.chainOfResponsibility;

public class FixedAmountDiscountHandler implements DiscountHandler {
    private DiscountHandler nextHandler;

    @Override
    public void setNextHandler(DiscountHandler nextHandler) {
        this.nextHandler = nextHandler;
    }

    @Override
    public void applyDiscount(Order order) {
        if (order.getTotalAmount() >= 50) {
            order.applyDiscount(5.0); // $5 discount
            System.out.println("Applied $5 discount.");
        } else if (nextHandler != null) {
            nextHandler.applyDiscount(order);
        }
    }
}
